prov/tcp/src/tcpx_progress.c

```diff
@@ -373,7 +373,7 @@ static void process_rx_pe_entry(struct tcpx_pe_entry *pe_entry)
 		}
 	}
 	if (pe_entry->state == TCPX_XFER_HDR_RECVD) {
-		field_offset += sizeof(pe_entry->msg_hdr);
+		field_offset = sizeof(pe_entry->msg_hdr);
```

Comment:
In this context this change looks okay, since there only two states. But the idea is to have offset zero initially and each state incrementally adds to it.  Same goes for the process_tx_pe_entry.

Reply:
so making offset zero initially would make more sense.

Reply:
I can change it to keep the += and instead initialize the offset at 0

Reply:
Awesome. Everything else looks good. Thanks

Reply:
I would keep the direct assignment.  If this function is re-entrant, you can't anticipate that the previous state checks have been run in the current function execution.  It would probably be better to convert the if (state check) to a switch-case statement, and let the code fall-through the case statements if the next state is reached.

Reply:
I think that is correct.  Offset has to be assigned in every if condition. Don't need to change this.

Reply:
A switch makes the most sense to me. Thoughts on the new changes @vkrishna ?